File: server-ruby-sinatra/api/rating_api.rb
require 'json'


MyApp.add_route('POST', '/api/{version}/rating/create', {
  "resourcePath" => "/Rating",
  "summary" => "Create Rating",
  "nickname" => "create_rating",
  "responseClass" => "RatingResponse",
  "endpoint" => "/api/{version}/rating/create",
  "notes" => "This is used to leave rating on a ratable object (i.e. retailer locations). Each user can only rate on a ratable object once per category. If a user rates on the same object and category, the previous rating will be overwritten. Leaving a rating on a ratable object will be visible to everyone who has access to view the object.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The unique device identifier that made the request (either deviceId or accountId must be used)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The unique accountId that made the request (either deviceId or accountId must be used)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ratable_type",
      "description" => "The ratable object type {RETAILER_LOCATION}",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ratable_id",
      "description" => "The id of the ratable object",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "rating_value",
      "description" => "The integer value of 0-100",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_id",
      "description" => "",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "display",
      "description" => "A subject title for the user rating (limited to 255 characters)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "The description of the rating",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_description",
      "description" => "The description of the location",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/rating/delete', {
  "resourcePath" => "/Rating",
  "summary" => "Delete Rating",
  "nickname" => "delete_rating",
  "responseClass" => "SirqulResponse",
  "endpoint" => "/api/{version}/rating/delete",
  "notes" => "Sets a rating as deleted.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The unique device identifier that made the request (either deviceId or accountId must be used)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The unique accountId that made the request (either deviceId or accountId must be used)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "rating_id",
      "description" => "The ID of the rating to delete",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/location/rating/index/search', {
  "resourcePath" => "/Rating",
  "summary" => "Search Location Rating Indexes",
  "nickname" => "search_location_rating_indexes",
  "responseClass" => "Array<RatingIndexResponse>",
  "endpoint" => "/api/{version}/location/rating/index/search",
  "notes" => "Search for retailer locations by averages near you.",
  "parameters" => [
    {
      "name" => "category_ids",
      "description" => "Comma separated list of category ids to filter the results by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "The keyword used to search",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_type",
      "description" => "The type of location to filter the results by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE}",
      "dataType" => "String",
      "allowableValues" => "[ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE]",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "The order to return the search results",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The record to begin the return set on",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The number of records to return",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "search_range",
      "description" => "The search radius in kilometers to filter results",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_overall_rating",
      "description" => "Determines whether to return the overall rating record instead",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "distance_unit",
      "description" => "",
      "dataType" => "String",
      "allowableValues" => "[MILES, KILOMETERS]",
      "paramType" => "query",
    },
    {
      "name" => "return_retailer",
      "description" => "whether to return the retailer or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_assets",
      "description" => "whether to return the assets or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_offers",
      "description" => "whether to return the offers or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_categories",
      "description" => "whether to return the categories or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_filters",
      "description" => "whether to return the filters or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/rating/index/search', {
  "resourcePath" => "/Rating",
  "summary" => "Search Rating Indexes",
  "nickname" => "search_rating_indexes",
  "responseClass" => "Array<RatingIndexResponse>",
  "endpoint" => "/api/{version}/rating/index/search",
  "notes" => "Search for ratable items by averages.",
  "parameters" => [
    {
      "name" => "ratable_type",
      "description" => "Filter results by a ratable type {RETAILER_LOCATION}",
      "dataType" => "String",
      "allowableValues" => "[LOCATABLE, RESERVABLE, PERMISSIONABLE, NOTABLE, ASSETABLE, LIKABLE, FLAGABLE, FAVORITABLE, RATABLE, ALBUM, COLLECTION, APPLICATION, APPLICATION_SETTING, APPLICATION_CERT, APPLICATION_PLACEMENT, ACCOUNT, ACCOUNT_SETTING, GAME_LEVEL, PACK, MISSION, TOURNAMENT, ASSET, ALBUM_CONTEST, THEME_DESCRIPTOR, OFFER, OFFER_LOCATION, EVENT, RETAILER, RETAILER_LOCATION, NOTE, CREATIVE, FAVORITE, LIKE, RATING, ANALYTIC, THIRD_PARTY_CREDENTIAL, THIRD_PARTY_NETWORK, REGION]",
      "paramType" => "query",
    },
    {
      "name" => "ratable_ids",
      "description" => "Comma separated list of ratable ids to filter the resuts by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_ids",
      "description" => "Comma separated list of category ids to filter the results by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "secondary_type",
      "description" => "",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "The keyword used to search",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE}",
      "dataType" => "String",
      "allowableValues" => "[ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE]",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "The order to return the search results",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The record to begin the return set on",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The number of records to return",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_ratable",
      "description" => "Determines whether to return the ratable object in the response",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "return_overall_rating",
      "description" => "Determines whether to return the overall rating record instead",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/{version}/rating/search', {
  "resourcePath" => "/Rating",
  "summary" => "Search Ratings",
  "nickname" => "search_ratings",
  "responseClass" => "Array<RatingResponse>",
  "endpoint" => "/api/{version}/rating/search",
  "notes" => "Search for ratings on a ratable object.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The device id (deviceId or accountId required)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The account id of the user (deviceId or accountId required)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "filter_account_id",
      "description" => "Filter results for a particular account",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ratable_type",
      "description" => "The ratable object type {RETAILER_LOCATION}",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "ratable_id",
      "description" => "The id of the ratable object",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_ids",
      "description" => "Comma separated list of category ids to filter the results by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "The keyword used to search",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, OWNER_DISPLAY, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, VALUE}",
      "dataType" => "String",
      "allowableValues" => "[ID, CREATED, UPDATED, DELETED, SEARCH_TAGS, ACTIVE, OWNER_DISPLAY, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, VALUE]",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "The order to return the search results",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The record to begin the return set on",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The number of records to return",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('POST', '/api/{version}/rating/update', {
  "resourcePath" => "/Rating",
  "summary" => "Update Rating",
  "nickname" => "update_rating",
  "responseClass" => "RatingResponse",
  "endpoint" => "/api/{version}/rating/update",
  "notes" => "Update an existing rating. Only the creator of the rating have permission to update.",
  "parameters" => [
    {
      "name" => "device_id",
      "description" => "The unique device identifier that made the request (either deviceId or accountId must be used)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "account_id",
      "description" => "The unique accountId that made the request (either deviceId or accountId must be used)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "rating_id",
      "description" => "The id of the rating (Note: this is not the ratable object id)",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "rating_value",
      "description" => "The rating value to update",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "category_id",
      "description" => "",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "display",
      "description" => "A subject title for the user rating (limited to 255 characters)",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "description",
      "description" => "The description of the rating",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "location_description",
      "description" => "The description of the location",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "latitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "longitude",
      "description" => "The current location of the user",
      "dataType" => "Float",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "version",
      "description" => "",
      "dataType" => "Float",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

File: server-scala-play-server/app/api/RatingApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.RatingIndexResponse
import model.RatingResponse
import model.SirqulResponse

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait RatingApi {
  /**
    * Create Rating
    * This is used to leave rating on a ratable object (i.e. retailer locations). Each user can only rate on a ratable object once per category. If a user rates on the same object and category, the previous rating will be overwritten. Leaving a rating on a ratable object will be visible to everyone who has access to view the object.
    * @param ratableType The ratable object type {RETAILER_LOCATION}
    * @param ratableId The id of the ratable object
    * @param ratingValue The integer value of 0-100
    * @param deviceId The unique device identifier that made the request (either deviceId or accountId must be used)
    * @param accountId The unique accountId that made the request (either deviceId or accountId must be used)
    * @param categoryId 
    * @param display A subject title for the user rating (limited to 255 characters)
    * @param description The description of the rating
    * @param locationDescription The description of the location
    * @param latitude The current location of the user
    * @param longitude The current location of the user
    */
  def createRating(version: BigDecimal, ratableType: String, ratableId: Long, ratingValue: Int, deviceId: Option[String], accountId: Option[Long], categoryId: Option[Long], display: Option[String], description: Option[String], locationDescription: Option[String], latitude: Option[Double], longitude: Option[Double]): RatingResponse

  /**
    * Delete Rating
    * Sets a rating as deleted.
    * @param ratingId The ID of the rating to delete
    * @param deviceId The unique device identifier that made the request (either deviceId or accountId must be used)
    * @param accountId The unique accountId that made the request (either deviceId or accountId must be used)
    */
  def deleteRating(version: BigDecimal, ratingId: Long, deviceId: Option[String], accountId: Option[Long]): SirqulResponse

  /**
    * Search Location Rating Indexes
    * Search for retailer locations by averages near you.
    * @param categoryIds Comma separated list of category ids to filter the results by
    * @param keyword The keyword used to search
    * @param locationType The type of location to filter the results by
    * @param sortField The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE}
    * @param descending The order to return the search results
    * @param start The record to begin the return set on
    * @param limit The number of records to return
    * @param searchRange The search radius in kilometers to filter results
    * @param latitude The current location of the user
    * @param longitude The current location of the user
    * @param returnOverallRating Determines whether to return the overall rating record instead
    * @param distanceUnit 
    * @param returnRetailer whether to return the retailer or not
    * @param returnAssets whether to return the assets or not
    * @param returnOffers whether to return the offers or not
    * @param returnCategories whether to return the categories or not
    * @param returnFilters whether to return the filters or not
    */
  def searchLocationRatingIndexes(version: BigDecimal, categoryIds: Option[String], keyword: Option[String], locationType: Option[String], sortField: Option[String], descending: Option[Boolean], start: Option[Int], limit: Option[Int], searchRange: Option[Double], latitude: Option[Double], longitude: Option[Double], returnOverallRating: Option[Boolean], distanceUnit: Option[String], returnRetailer: Option[Boolean], returnAssets: Option[Boolean], returnOffers: Option[Boolean], returnCategories: Option[Boolean], returnFilters: Option[Boolean]): List[RatingIndexResponse]

  /**
    * Search Rating Indexes
    * Search for ratable items by averages.
    * @param ratableType Filter results by a ratable type {RETAILER_LOCATION}
    * @param ratableIds Comma separated list of ratable ids to filter the resuts by
    * @param categoryIds Comma separated list of category ids to filter the results by
    * @param secondaryType 
    * @param keyword The keyword used to search
    * @param sortField The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, COUNT, SUMMATION, AVERAGE, VALUE}
    * @param descending The order to return the search results
    * @param start The record to begin the return set on
    * @param limit The number of records to return
    * @param latitude 
    * @param longitude 
    * @param returnRatable Determines whether to return the ratable object in the response
    * @param returnOverallRating Determines whether to return the overall rating record instead
    */
  def searchRatingIndexes(version: BigDecimal, ratableType: String, ratableIds: Option[String], categoryIds: Option[String], secondaryType: Option[String], keyword: Option[String], sortField: Option[String], descending: Option[Boolean], start: Option[Int], limit: Option[Int], latitude: Option[Double], longitude: Option[Double], returnRatable: Option[Boolean], returnOverallRating: Option[Boolean]): List[RatingIndexResponse]

  /**
    * Search Ratings
    * Search for ratings on a ratable object.
    * @param deviceId The device id (deviceId or accountId required)
    * @param accountId The account id of the user (deviceId or accountId required)
    * @param filterAccountId Filter results for a particular account
    * @param ratableType The ratable object type {RETAILER_LOCATION}
    * @param ratableId The id of the ratable object
    * @param categoryIds Comma separated list of category ids to filter the results by
    * @param keyword The keyword used to search
    * @param sortField The column to sort the search on {ID, CREATED, UPDATED, ACTIVE, OWNER_DISPLAY, RATABLE_TYPE, RATABLE_ID, RATABLE_DISPLAY, CATEGORY_ID, CATEGORY_NAME, CATEGORY_SHORT_NAME, CATEGORY_DISPLAY, VALUE}
    * @param descending The order to return the search results
    * @param start The record to begin the return set on
    * @param limit The number of records to return
    */
  def searchRatings(version: BigDecimal, deviceId: Option[String], accountId: Option[Long], filterAccountId: Option[Long], ratableType: Option[String], ratableId: Option[Long], categoryIds: Option[String], keyword: Option[String], sortField: Option[String], descending: Option[Boolean], start: Option[Int], limit: Option[Int]): List[RatingResponse]

  /**
    * Update Rating
    * Update an existing rating. Only the creator of the rating have permission to update.
    * @param ratingId The id of the rating (Note: this is not the ratable object id)
    * @param deviceId The unique device identifier that made the request (either deviceId or accountId must be used)
    * @param accountId The unique accountId that made the request (either deviceId or accountId must be used)
    * @param ratingValue The rating value to update
    * @param categoryId 
    * @param display A subject title for the user rating (limited to 255 characters)
    * @param description The description of the rating
    * @param locationDescription The description of the location
    * @param latitude The current location of the user
    * @param longitude The current location of the user
    */
  def updateRating(version: BigDecimal, ratingId: Long, deviceId: Option[String], accountId: Option[Long], ratingValue: Option[Int], categoryId: Option[Long], display: Option[String], description: Option[String], locationDescription: Option[String], latitude: Option[Double], longitude: Option[Double]): RatingResponse
}
